_posts/2019-06-24-logging-call-into-your-service-with-ringcentral-embeddable.md

```diff
@@ -0,0 +1,153 @@
+  type: "OnDemand"
+  uri: "https://platform.devtest.ringcentral.com/restapi/v1.0/account/170848004/recording/6469338004"
+}
+```
+
+The `link` property in `recording` is a link to get and play recording file from RingCentral server. The `contentUri` is a URI which can be used to get `recording` file with RingCentral access token. If you pass `recordingWithToken` when register service, you can get contentUri with access_token. The access_token will be expired in minutes, so need to download immediately when get it.
+
+```js
+{
+  contentUri: "https://media.devtest.ringcentral.com/restapi/v1.0/account/170848004/recording/6469338004/content?access_token=ringcentral_access_token"
+  id: "6469338004"
+  link: "http://apps.ringcentral.com/integrations/recording/sandbox/?id=Ab7937-59r6EzUA&recordingId=6469338004"
+  type: "OnDemand"
+  uri: "https://platform.devtest.ringcentral.com/restapi/v1.0/account/170848004/recording/6469338004"
+}
+```
+
+In this section, we can log the call into your service directly, or popup a window for user to confirm before saved into your service.
+
+## Add call log entity matcher
+
+The widget needs to know if a call is logged, so we need to provide `matcher` response:
+
+```js
+document.querySelector("#rc-widget-adapter-frame").contentWindow.postMessage({
+  type: 'rc-adapter-register-third-party-service',
+  service: {
+    name: 'TestService',
+    callLoggerPath: '/callLogger',
+    callLoggerTitle: 'Log to TestService',
+    callLogEntityMatcherPath: '/callLogger/match',
+  }
+}, '*');
+```
+
+If we provide `callLogEntityMatcherPath` when we register third party service, the widget will send call matcher request when app is loaded or call is logged. 
+
+```js
+window.addEventListener('message', function (e) {
+  var data = e.data;
+  if (data && data.type === 'rc-post-message-request') {
+    if (data.path === '/callLogger/match') {
+      // add your codes here to reponse match result
+      console.log(data); // get call session id list in here
+      // response to widget
+      document.querySelector("#rc-widget-adapter-frame").contentWindow.postMessage({
+        type: 'rc-post-message-response',
+        responseId: data.requestId,
+        response: {
+          data: {
+            '214705503020': [{ // call session id from request
+              id: '88888', // call log entity id from your platform
+              note: 'Note', // Note of this call log entity
+            }]
+          }
+        },
+      }, '*');
+    }
+  }
+});
+```
+
+If the widget gets call logger entity from the matcher service, it will show green logger button.
+
+## Conclusion
+
+By here, we have finished the call logging feature. We just need to register the service and respond to the widget's message to add your customize code. Hope this article will help you to integrate call logging feature with RingCentral Embeddable quickly.
```

Comment:
`At this point`, we have finished the call logging feature. We just need to register the service and respond to the widget's `messages` to add your `customization`. `I` hope this article will...